pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -127,7 +151,15 @@ Map<String, Operation> computeOperation(SegmentDirectory.Reader segmentReader)
     Set<String> newNoDictColumns = _indexLoadingConfig.getNoDictionaryColumns();
 
     for (String column : existingAllColumns) {
-      if (existingNoDictColumns.contains(column) && newNoDictColumns.contains(column)) {
+      if (existingNoDictColumns.contains(column) && !newNoDictColumns.contains(column)) {
+        if (_schema == null || _indexLoadingConfig.getTableConfig() == null) {
+          // This can only happen in tests.
```

Comment:
why should this happen in tests ? I feel this should be a preconditions check

Reply:
There are some tests that call SegmentPreProcessor with null schema. I added this condition because those tests could end up failing.